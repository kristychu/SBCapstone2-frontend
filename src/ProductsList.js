import React from "react";
import { useSelector, useDispatch } from "react-redux";
import Product from "./Product";
import { Card, CardText, CardBody, Button } from "reactstrap";
import { createStep } from "./actions";

function ProductsList({ routineStep, timeOfDay, toggle }) {
  const token = useSelector((st) => st.users.profile.token);
  const decoded = JSON.parse(atob(token.split(".")[1]));
  const username = decoded.username;

  const dispatch = useDispatch();
  const products = useSelector((st) => st.productResults);

  const addProduct = (productId) => {
    dispatch(createStep(username, token, routineStep, timeOfDay, productId));
    toggle();
  };

  return (
    <div>
      {products.length > 0 && <p>{products.length} Results:</p>}
      {products.length > 0
        ? products.map(({ id, brand, name }) => (
            <div key={id} id={products[id]}>
              <Card>
                {/* <CardImg top width="100%" src="/assets/318x180.svg" alt="Card image cap" /> */}
                <CardBody>
                  <CardText>
                    <Product id={id} brand={brand} name={name} />
                  </CardText>
                  <Button color="success" onClick={() => addProduct(id)}>
                    Add
                  </Button>
                </CardBody>
              </Card>
            </div>
          ))
        : `Search for something!`}
    </div>
  );
}

export default ProductsList;
