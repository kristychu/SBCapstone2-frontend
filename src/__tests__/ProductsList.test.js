import React from "react";
import { Provider } from "react-redux";
import { createStore } from "redux";
import reducer from "../reducers/rootReducer";
import { render } from "@testing-library/react";
import ProductsList from "../ProductsList";
import { MemoryRouter } from "react-router";

const store = createStore(reducer);

it("should render without crashing", () => {
  render(
    <Provider store={store}>
      <MemoryRouter>
        <ProductsList />
      </MemoryRouter>
    </Provider>
  );
});

it("should match snapshot", function () {
  const { asFragment } = render(
    <Provider store={store}>
      <MemoryRouter>
        <ProductsList />
      </MemoryRouter>
    </Provider>
  );
  expect(asFragment()).toMatchSnapshot();
});
