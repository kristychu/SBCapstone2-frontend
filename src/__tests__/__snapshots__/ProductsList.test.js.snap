// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match snapshot 1`] = `
<DocumentFragment>
  <div>
    <p>
      You need an account to access this page.
    </p>
    <p>
      Have an account already? 
      <a
        href="/login"
      >
        Login!
      </a>
    </p>
    <p>
      Otherwise, 
      <a
        href="/register"
      >
        Register here
      </a>
      .
    </p>
  </div>
</DocumentFragment>
`;
